### PR TITLE
icon improvements for waveform and rgb parade

### PR DESCRIPTION
**Pull request by dtorop**
Make the "scope type" waveform icon be transparent, to work better with the theme work done by @Nilvus. Also tune its drawing slightly, as transparency made the prior drawing code look less good.

Change the "waveform type" icons for "overlaid" (e.g. traditional waveform) and rgb parade from solid colors to more waveform-ish looking images.

This PR includes the three CSS fix commits from #7952 in order to be able to see all the changes together.

Prior to #7952:

image

image

With #7952 and without this PR:

image

image

With #7952 and this PR:

image

image

### CODE REVIEW

**Comment by TurboGit** [comment]
This looks good to me. I'll let @Nilvus review before merging.

**Comment by dtorop** [comment]
Rebased this to master, now that #7952 is merged.

**Comment by johnny-bit** [comment]
I'd say that colours are bit too saturated on last 4 buttons... the transparency in buttons "prior to #7952" look nice... that being said I haven't tested "the look" just checked screenshots ;)

**Comment by Nilvus** [comment]
> I'd say that colours are bit too saturated on last 4 buttons... the transparency in buttons "prior to #7952" look nice... that being said I haven't tested "the look" just checked screenshots ;)

That's screenshots and it's not related too that PR but my mine (only 3 channels one), merged now (#7952). They are at 66% of red, green or blue full color, so not really saturated.

I will test that PR probably tomorrow evening. Not possible before.

**Comment by dtorop** [comment]
> I will test that PR probably tomorrow evening. Not possible before.

Sounds good!

> I'd say that colours are bit too saturated on last 4 buttons...

As @Nilvus relates, that goes back to #7952 rather than this PR. But for context, here are the buttons in RGB parade, with the whole scope visible, in three themes:

image

image

image

**Comment by todd-prior** [comment]
I know there has been a ton of work on this and likely I am just too naive to understand but I was wondering why to we keep the parade display so muted and washed out. Maybe its my old eyes but I rarely use the parades because  at least on my monitor it a strain to use them. It may be set this way so as not to affect the editing process which from a technical point I would totally get. Its much like the grey UI is obviously superior for editing but not very pretty. I recall a while back someone mentioned the alpha channel setting used. Is there a way a user could hack that to bump brightness and contrast?? I am not asking for a change just sharing my experience.....as always thanks for the hard work to make things better....

**Comment by dtorop** [comment]
> I recall a while back someone mentioned the alpha channel setting used. Is there a way a user could hack that to bump brightness and contrast

It's a fair question -- I've wondered too. I think that the alpha value was built into the histogram, where it gave it a good look -- and as the histogram is a constant lightness, it was legible. The alpha remained in waveform. It made the waveform less legible, as it has varying density, but gave a consistent look. Then with RGB parade it's even harder, as each channel is only ~1/3 brightness of the waveform.

It's worth re-examining (outside of this PR). I think the alpha is hardcoded, but best case it could make its way into CSS and this could be a design question... I hate to suggest, but maybe a bug report or feature request about this would be good?

**Comment by Nilvus** [comment]
> I know there has been a ton of work on this and likely I am just too naive to understand but I was wondering why do we keep the parade display so muted and washed out. Maybe its my old eyes but I rarely use the parades because at least on my monitor its a strain to use them.

I agree, a little less washed out would be better but indeed, it's not related to that PR as this one is just about icons.

> Its much like the grey UI is obviously superior for editing but not very pretty.

Not about that PR but could you precise on what? Pretty is quite subjective and imprecise.

> I recall a while back someone mentioned the alpha channel setting used. Is there a way a user could hack that to bump brightness and contrast?? I am not asking for a change just sharing my experience.....as always thanks for the hard work to make things better....

Yep, that would be good.

**Review by Nilvus — APPROVED** [review]
It's all ok for me. And hover effect works great. Thanks @dtorop. Could be merged for me.

So next step (with another PR) would be to allow better rgb parade and waveform as @todd-prior suggest.

**Comment by todd-prior** [comment]
Using menus and dialog boxes in a UI is usually nicer when things are bright colorful and “contrasty” not the best for editing …that is what I was alluding to zi was not advocating for change there.

**Comment by Nilvus** [comment]
> Using menus and dialog boxes in a UI is usually nicer when things are bright colorful and “contrasty” not the best for editing …that is what I was alluding to zi was not advocating for change there. 

Oh ok. Of course, better is contrasted, better is it. And on grey theme, I set the best contrast is possible when using middle grey theme. Not possible to do better. Thanks for your reply.

**Comment by dtorop** [comment]
> It's all ok for me. And hover effect works great. Thanks @dtorop. Could be merged for me.

Great! Thank you for reviewing. There can always be another pass for icons, but this is an improvement on what is in git master.

> So next step (with another PR) would be to allow better rgb parade and waveform as @todd-prior suggest.

Yes... I'm happy to look into this -- ideal would be a way to control it via CSS. I'm (finally) back to vectorscope work, so will try to get a basic version of that into a PR first. While doing this work, I've realized that I can do things with buttons and waveform gamma better, hence there may still be some general improvements.

**Review by TurboGit — APPROVED** [review]
Looks good to me, thanks for working on this!

**Pull request merged**
